make-base/stubs.mk

```diff
@@ -34,35 +46,42 @@ bash:: ## Starts a bash shell in an app environment for the given project
 psql:: ## Starts a psql shell in an app environment for the given project
 	@ true
 
-.PHONY: migrate
-migrate:: ## Runs migrations in an app environment for the given project
-	@ true
-
 .PHONY: restart
 restart:: ## Restarts all containers for the given project(s)
 	@ true
 
 .PHONY: release
-release:: ## Builds static docker images staged for release, runs tests against them, and pushes them to dockerhub (requires PUSH=1)
+release:: release-build release-test ## Builds static docker images staged for release, runs tests against them, and pushes them to dockerhub (requires PUSH_IMAGES=1)
 	@ true
 
 .PHONY: release-build
-release-build:: config-ready
+release-build:: config-ready ## Step that prepares a release build if necessary
 	@ true
 
-.PHONY: release-test
-release-test::
+.PHONY: release-test ## Step that prepares a release test if necessary
+release-test:: config-ready release-build
 	@ true
 
-.PHONY: irb
-irb:: ## Starts up an irb session in the context of the given project
+.PHONY: config-ready
+config-ready:: ## Setup step that ensures that configuration files necessary for preparation of the development environment are ready.
 	@ true
 
-config-ready::
+.PHONY: update
+update:: update-ready ## Step to force update of database and development dependencies.
 	@ true
 
-update::
+.PHONY: update-ready
+update-ready::
 	@ true
 
-docker-ready::
+.PHONY: compose-ready
+compose-ready:: ## Setup step that ensures that the images are ready to run the development docker-compose.yml file
 	@ true
+
+.PHONY: _test-find-all
+_test-find-all:
+	@ echo $(call map,dirname,$(call find_projects_containing,Makefile))
+
+ifneq ($(wildcard ./Dockerfile),)
+$(eval $(call image_target_here))
```

Comment:
Whenever `stubs.mk` is included in a project `Makefile` it will automatically sniff for a `Dockerfile` and create some relevant build targets for it as well.